src/Geocoder/Model/AddressFactory.php

```diff
@@ -20,6 +22,18 @@ public function createFromArray(array $results)
     {
         $addresses = [];
         foreach ($results as $result) {
+            $adminLevels = [];
+            foreach ($this->readArrayValue($result, 'adminLevels') as $level => $adminLevel) {
+                $level = $this->normalizeLevelValue($level);
+                $adminLevels[$level] = new AdminLevel(
+                    $level,
+                    $this->readStringValue($adminLevel, 'name'),
+                    $this->readStringValue($adminLevel, 'code')
+                );
+            }
+
+            ksort($adminLevels, SORT_NUMERIC);
```

Comment:
I think I've added a test which expose this, but here the PHP magic to solve:
```PHP
$admins = [];
$admins[2] = 'admin 2';
$admins[1] = 'admin 1';

echo implode(', ', $admins) . "\n";
foreach ($admins as $level => $admin) {
    echo $level . ': ' . $admin . "\n";
}
```

expected:
```
admin1, admin2
1: admin 1
2: admin 2
```

actual:
```
admin2, admin1
2: admin 2
1: admin 1
```

Since Providers deal with external data, I think is much safer to sort it out explicitely. Anyway, all of this is solved by introducing the `AdminLevelCollection` class.